### PR TITLE
Debt API

### PR DESCRIPTION
**Pull request by lizhengwei88**
Debt API

### CODE REVIEW

File: api/routers/router.go [hunk 1]
```diff
@@ -49,7 +49,8 @@ func (r *Router) Init(e *gin.Engine) {
 	v1.GET("/Homeaccounts", r.AccountHandler.GetHomeAccounts())
 	v1.GET("/txs", r.BlockHandler.GetTxs())
 	v1.GET("/pendingtxs", r.BlockHandler.GetPendingTxs())
-	v1.GET("blockTxsTps", r.BlockHandler.GetBlockTxsTps())
+	v1.GET("/debts", r.BlockHandler.Getdebts())
+	v1.GET("/blockTxsTps", r.BlockHandler.GetBlockTxsTps())
```
Comment (WTer) [inline api/routers/router.go]: nice to sort this path alphabetically

File: database/client.go [hunk 2]
```diff
@@ -496,6 +506,21 @@ func (c *Client) GetTxCntByShardNumber(shardNumber int) (uint64, error) {
 	return txCnt, err
 }
 
+//GetdebtCntByShardNumber get tx count by shardNumber
+func (c *Client) GetdebtCntByShardNumber(shardNumber int) (uint64, error) {
+	var txCnt uint64
+	query := func(c *mgo.Collection) error {
+		var err error
+		//TODO: fix this overflow
```
Comment (WTer) [inline database/client.go]: who will be overflow?

File: api/routers/router.go (hunk 1 above)
Comment (lizhengwei88) [reply api/routers/router.go]: OK

File: database/client.go (hunk 2 above)
Comment (lizhengwei88) [reply database/client.go]: OK

File: api/handlers/blockinfo_handler.go [hunk 3]
```diff
@@ -771,6 +790,71 @@ func (h *BlockHandler) GetTxs() gin.HandlerFunc {
...
+		ps, _ := strconv.ParseUint(c.Query("ps"), 10, 64)
+		s, _ := strconv.ParseInt(c.Query("s"), 10, 64)
+		if ps == 0 {
+			ps = transItemNumsPrePage
+		} else if ps > maxItemNumsPrePage {
+			ps = maxItemNumsPrePage
+		}
+
+		if p >= 1 {
+			p--
```
Comment (WTer) [inline api/handlers/blockinfo_handler.go]: why --?
add comments here

File: api/handlers/blockinfo_handler.go [hunk 4]
```diff
@@ -771,6 +790,71 @@ func (h *BlockHandler) GetTxs() gin.HandlerFunc {
...
+		if s <= 0 {
+			s = 1
+		}
+		shardNumber := int(s)
+
+		block, flag := c.GetQuery("block")
+		if flag {
+			height, err := strconv.ParseUint(block, 10, 64)
+			if err != nil {
+				responseError(c, errParamInvalid, http.StatusBadRequest, apiParmaInvalid)
```
Comment (WTer) [inline api/handlers/blockinfo_handler.go]: need to add return after it?

File: api/handlers/blockinfo_handler.go (hunk 3 above)
Comment (lizhengwei88) [reply api/handlers/blockinfo_handler.go]: By the total number according to the article and ps distribution calculated totalPages, conversion
	if page > (totalPages - 1) {
		page = totalPages - 1
	}
	end = total - page*step

File: api/handlers/blockinfo_handler.go (hunk 4 above)
Comment (lizhengwei88) [reply api/handlers/blockinfo_handler.go]: OK

**Pull request merged**